File: design-patterns-2/ts/typescript/patterns/factories/factory/Runner.ts
import IRunner from "../../IRunner";
import CarFactory from "./CarFactory";

export default class Runner implements IRunner{
    run(exampleNumber: number=1){
        const carFactory = new CarFactory();

        const car = carFactory.createCar('Hatchback', 'some model', 2007);
        car?.displayCarInfo();
    }
}

File: design-patterns-2/csharp/patterns/proxy/Runner.cs
using csharp.patterns.proxy.protectionProxy;

namespace csharp.patterns.proxy;

class Runner : IRunner
{
    public void Run(int exampleNumber)
    {
        if (exampleNumber == 1)
        {
            //Protection proxy, you can not drive if you are under 18 yo
            ICar car = new CarProxy(new Driver(5));
            car.Drive();
        }
    }
}